pir/pir-internal/src/main/java/com/duckduckgo/pir/internal/scripts/PirMessagingInterface.kt

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2025 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.pir.internal.scripts
+
+import android.webkit.JavascriptInterface
+import android.webkit.WebView
+import com.duckduckgo.di.scopes.ServiceScope
+import com.duckduckgo.js.messaging.api.JsCallbackData
+import com.duckduckgo.js.messaging.api.JsMessage
+import com.duckduckgo.js.messaging.api.JsMessageCallback
+import com.duckduckgo.js.messaging.api.JsMessageHandler
+import com.duckduckgo.js.messaging.api.JsMessageHelper
+import com.duckduckgo.js.messaging.api.JsMessaging
+import com.duckduckgo.js.messaging.api.JsRequestResponse
+import com.duckduckgo.js.messaging.api.SubscriptionEvent
+import com.duckduckgo.js.messaging.api.SubscriptionEventData
+import com.duckduckgo.pir.internal.brokers.JSONObjectAdapter
+import com.squareup.anvil.annotations.ContributesBinding
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import javax.inject.Inject
+import javax.inject.Named
+import logcat.logcat
+
+@ContributesBinding(
+    scope = ServiceScope::class,
+    boundType = JsMessaging::class,
+)
+@Named("BrokerProtection")
```

Comment:
Didn't see where it is used, but should it rather be something like
```kotlin
@Retention(AnnotationRetention.BINARY)
@Qualifier
private annotation class BrokerProtection
```

to avoid the string?

Reply:
Only used in a single place. I could add this for sure, doesn’t do any harm i think.

Reply:
Will update this later. I will remove this anotation altogether